### PR TITLE
Improve creating expenses on Collect with Instant Submit

### PR DESCRIPTION
**Pull request by youssef-lr**
### Details

~Held on https://github.com/Expensify/Auth/pull/9916~

cc @Beamanator 

### Fixed Issues

$ https://github.com/Expensify/App/issues/34955

### Tests

- [ ] Verify that no errors appear in the JS console

1. Create a Collect policy in OldDot, verify it has the `autoReportingFrequency` set to `instant`, if it doesn't, run the following in the policy page as the admin: `const p = Policy.getCurrent(); p.policy.autoReportingFrequency = 'instant'; p.save();`
2. Invite a member to the workspace `submitter@gmail.com`.
3. Invite another member `approver@gmail.com` to the workspace, set it as the default approver.
4. Log in as the submitter in NewDot, and create your first expense report in the workspace chat.
5. Create another request and make sure it goes to the same report preview.
6. Click on the report preview to navigate to the expense report.
7. Click on the '+' and make sure you see `Request Money`, click on it and make sure the request gets added to the current report.
8. Approve the report as the approver.
9. Create another request, and make sure this time it goes to a new report.
10. Repeat these steps using the Request Money Scan tab, Manual Split, Scan Split, & Distance. 

### Offline tests

The same steps as above, except, go offline before requesting money, then go back online. Make sure everything looks right as described in the s

### QA Steps

- [ ] Verify that no errors appear in the JS console

1. Create a Collect policy in OldDot, verify it has the `autoReportingFrequency` set to `instant`, if it doesn't, run the following in the policy page as the admin: `const p = Policy.getCurrent(); p.policy.autoReportingFrequency = 'instant'; p.save();`
2. Invite a member to the workspace `submitter@gmail.com`.
3. Invite another member `approver@gmail.com` to the workspace, set it as the default approver.
4. Log in as the submitter in NewDot, and create your first expense report in the workspace chat.
5. Create another request and make sure it goes to the same report preview.
6. Click on the report preview to navigate to the expense report.
7. Click on the '+' and make sure you see `Request Money`, click on it and make sure the request gets added to the current report.
8. Approve the report as the approver.
9. Create another request, and make sure this time it goes to a new report.
10. Repeat these steps using the Request Money Scan tab, Manual Split, Scan Split, & Distance. 

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [X] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/9680864/78fc567c-2b0b-4766-8801-6bfd8096780a

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

File: src/libs/actions/IOU.ts [hunk 1]
```diff
@@ -1728,7 +1735,7 @@ function createSplitsAndOnyxData(
         // For Control policy expense chats, if the report is already approved, create a new expense report
         let oneOnOneIOUReport: OneOnOneIOUReport = oneOnOneChatReport.iouReportID ? allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${oneOnOneChatReport.iouReportID}`] : null;
         const shouldCreateNewOneOnOneIOUReport =
-            !oneOnOneIOUReport || (isOwnPolicyExpenseChat && ReportUtils.isControlPolicyExpenseReport(oneOnOneIOUReport) && ReportUtils.isReportApproved(oneOnOneIOUReport));
+            !oneOnOneIOUReport || (isOwnPolicyExpenseChat && ReportUtils.isPaidGroupPolicy(oneOnOneIOUReport) && ReportUtils.isReportApproved(oneOnOneIOUReport));
```
[the remark this answers is not in the document]
Comment (youssef-lr) [reply src/libs/actions/IOU.ts:1738]: We previously only supported Control type of workspaces for splits. This logic checks whether or not we can add a new expense (designated IOU here) which is created from a split to an existing report or create a new one. If the report is approved (approvals only exist in paid policies, hence the `isPaidGroupPolicy` check), we add the new expense to a new report. Makes sense?

[the remark this answers is not in the document]
Comment (youssef-lr) [reply src/libs/actions/IOU.ts:1738]: >Just wanna double check whether we should use ReportUtils.isPaidGroupPolicyExpenseReport instead of ReportUtils.isPaidGroupPolicy here.

It doesn't really matter, both should work fine.

File: src/libs/ReportUtils.ts [hunk 2]
```diff
@@ -4255,7 +4255,7 @@ function canRequestMoney(report: OnyxEntry<Report>, policy: OnyxEntry<Policy>, o
     if (isMoneyRequestReport(report)) {
         const isOwnExpenseReport = isExpenseReport(report) && isOwnPolicyExpenseChat;
         if (isOwnExpenseReport && PolicyUtils.isPaidGroupPolicy(policy)) {
-            return isDraftExpenseReport(report);
+            return isDraftExpenseReport(report) || (PolicyUtils.isInstantSubmitEnabled(policy) && isProcessingReport(report));
```
[the remark this answers is not in the document]
Comment (youssef-lr) [reply src/libs/ReportUtils.ts:4258]: I think this [comment](https://github.com/Expensify/App/blob/4eb5cbffbf0c8c37829b484a9c61bc58bf3f62c5/src/libs/ReportUtils.ts#L4222) already explains this logic. If Instant Submit is turned on, we can add expenses to an already submitted report. If reporting frequency is set to 'Weekly', then we can only add expenses to draft reports (i.e. not submitted yet).

**Comment by youssef-lr** [comment]
Thanks!

**Review by hoangzinh — APPROVED** [review]
LGTM

**Comment by Beamanator** [comment]
Faaak - this PR has a bit of overlap with another PR - see my comments in slack [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1709044956498699)

File: src/libs/ReportUtils.ts (hunk 2 above)
Comment (Beamanator) [inline src/libs/ReportUtils.ts:4258]: I believe at this point you can revert this change after pulling `main` - since this was just added in https://github.com/Expensify/App/pull/36997 - sorry!

File: src/libs/ReportUtils.ts [hunk 3]
```diff
@@ -2921,11 +2921,11 @@ function buildOptimisticExpenseReport(chatReportID: string, policyID: string, pa
     const formattedTotal = CurrencyUtils.convertToDisplayString(storedTotal, currency);
     const policy = getPolicy(policyID);
 
-    const isFree = policy?.type === CONST.POLICY.TYPE.FREE;
+    const isInstantSubmitEnabled = policy?.type === CONST.POLICY.TYPE.FREE || (PolicyUtils.isPaidGroupPolicy(policy) && PolicyUtils.isInstantSubmitEnabled(policy));
```
Comment (Beamanator) [inline src/libs/ReportUtils.ts:2924]: Now you might be able to reuse the new `ReportUtils.isExpenseReportWithInstantSubmittedState` here?

File: src/libs/ReportUtils.ts (hunk 2 above)
Comment (youssef-lr) [reply src/libs/ReportUtils.ts:4258]: @Beamanator I'm thinking of removing that helper function, it seems unnecessary and the naming is verbose and a bit confusing. What do you think?

Comment (youssef-lr) [reply src/libs/ReportUtils.ts:4258]: What do you think?

File: src/libs/actions/IOU.ts [hunk 4]
```diff
@@ -758,10 +758,17 @@ function getMoneyRequestInformation(
         // If the scheduled submit is turned off on the policy, user needs to manually submit the report which is indicated by GBR in LHN
         needsToBeManuallySubmitted = isFromPaidPolicy && !policy?.harvesting?.enabled;
 
-        // If the linked expense report on paid policy is not draft, we need to create a new draft expense report
-        if (iouReport && isFromPaidPolicy && !ReportUtils.isDraftExpenseReport(iouReport)) {
+        const isProcessingReportOnInstantSubmitPolicy =
+            isFromPaidPolicy && ReportUtils.isProcessingReport(iouReport) && PolicyUtils.isInstantSubmitEnabled(policy ?? ({} as OnyxEntry<OnyxTypes.Policy>));
```
[the remark this answers is not in the document]
Comment (youssef-lr) [reply src/libs/actions/IOU.ts:762]: Replied [here](https://github.com/Expensify/App/pull/36388/files#r1505230327)! Let me know your thoughts

File: src/libs/ReportUtils.ts (hunk 2 above)
Comment (Beamanator) [reply src/libs/ReportUtils.ts:4258]: I see what you mean & yeah maybe the function name is kinda gross, but don't we reuse the same check (`PolicyUtils.isInstantSubmitEnabled(policy) && isProcessingReport(report)`) in many places & won't we continue to do so? So wouldn't it be nice to make a util function for this? I slightly prefer a util function IFF we can come up with a good name for it that's super clear 😅

Comment (youssef-lr) [reply src/libs/ReportUtils.ts:4258]: Sounds good, I can't think of a way to improve that function name, so I think I'll create a new one which will instead decide if we can add transactions to a report or not, similar to how we do it in Auth. Will commit the update and let me know what you think! I think it looks cleaner.

File: src/components/MoneyRequestHeader.tsx [hunk 5]
```diff
@@ -79,16 +79,11 @@ function MoneyRequestHeader({session, parentReport, report, parentReportAction,
     const isScanning = TransactionUtils.hasReceipt(transaction) && TransactionUtils.isReceiptBeingScanned(transaction);
     const isPending = TransactionUtils.isExpensifyCardTransaction(transaction) && TransactionUtils.isPending(transaction);
 
-    const isRequestModifiable = !isSettled && !isApproved && !ReportActionsUtils.isDeletedAction(parentReportAction);
-    const canModifyRequest = isActionOwner && !isSettled && !isApproved && !ReportActionsUtils.isDeletedAction(parentReportAction);
```
Comment (youssef-lr) [inline src/components/MoneyRequestHeader.tsx:83]: I also removed this one, because we aren't really modifying the request in the header. We just need to check for permission to hold/unhold, and delete.

File: src/components/MoneyRequestHeader.tsx [hunk 6]
```diff
@@ -79,16 +79,11 @@ function MoneyRequestHeader({session, parentReport, report, parentReportAction,
     const isScanning = TransactionUtils.hasReceipt(transaction) && TransactionUtils.isReceiptBeingScanned(transaction);
     const isPending = TransactionUtils.isExpensifyCardTransaction(transaction) && TransactionUtils.isPending(transaction);
 
-    const isRequestModifiable = !isSettled && !isApproved && !ReportActionsUtils.isDeletedAction(parentReportAction);
```
Comment (youssef-lr) [inline src/components/MoneyRequestHeader.tsx:82]: I've removed this as it's confusing while we already have `canModifyRequest`. Also, this is only used to decide if user can hold or unhold the request, so I renamed it accordingly.

File: src/components/MoneyRequestHeader.tsx [hunk 7]
```diff
@@ -79,16 +79,11 @@ function MoneyRequestHeader({session, parentReport, report, parentReportAction,
...
-    const canModifyRequest = isActionOwner && !isSettled && !isApproved && !ReportActionsUtils.isDeletedAction(parentReportAction);
-    let canDeleteRequest = canModifyRequest;
-
-    if (ReportUtils.isPaidGroupPolicyExpenseReport(moneyRequestReport)) {
-        // If it's a paid policy expense report, only allow deleting the request if it's in draft state or instantly submitted state or the user is the policy admin
-        canDeleteRequest =
-            canDeleteRequest &&
```
Comment (youssef-lr) [inline src/components/MoneyRequestHeader.tsx:89]: Confirmed in Slack that the admin should not be able to delete requests, the backend also throws an error if we try to. This check was also never true `PolicyUtils.isPolicyAdmin(policy));` , as `canDeleteRequest` assumes the actor is the owner of the action, so it contradicted `isPolicyAdmin`.

File: src/libs/ReportUtils.ts [hunk 8]
```diff
@@ -4977,6 +4987,17 @@ function canBeAutoReimbursed(report: OnyxEntry<Report>, policy: OnyxEntry<Policy
     return isAutoReimbursable;
 }
 
+/**
+ * Used from money request actions to decide if we need to build an optimistic money request report.
+   Create a new report if:
+   - we don't have an iouReport set in the chatReport
+   - we have one, but it's waiting on the payee adding a bank account
+   - we have one but we can't add more transactions to it due to: report is approved or settled, or, report is processing and policy isn't under Instant Submit
```
Comment (Beamanator) [inline src/libs/ReportUtils.ts:4995]:
```suggestion
   - we have one but we can't add more transactions to it due to: report is approved or settled, or report is processing and policy isn't on Instant submit reporting frequency
```

File: src/libs/ReportUtils.ts [hunk 9]
```diff
@@ -1244,6 +1236,29 @@ function getChildReportNotificationPreference(reportAction: OnyxEntry<ReportActi
     return isActionCreator(reportAction) ? CONST.REPORT.NOTIFICATION_PREFERENCE.ALWAYS : CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN;
 }
 
+/**
+ * Checks whether the supplied report supports adding more transactions to it.
+ * Return true if:
+ * - report is a non-settled IOU
+ * - report is a draft
+ * - report is processing and policy's on Instant Submit
```
Comment (Beamanator) [inline src/libs/ReportUtils.ts:1244]:
```suggestion
 * - report is a processing expense report and its policy has Instant reporting frequency
```

**Review by Beamanator — COMMENTED** [review]
Ohhhhh baby, I think I absolutely love your changes, but there's a lot of conditionals that have changed 😅  So I think we're going to need some more rigorous testing before we can merge this, what you think?

**Comment by youssef-lr** [comment]
>So I think we're going to need some more rigorous testing before we can merge this, what you think?

Yes, though I'm gonna double check everything before pingin @hoangzinh for another re-review, I'll be extra careful :)

**Comment by youssef-lr** [comment]
Actually, looks like we already have unit tests for this, I'll be updating them.

**Comment by youssef-lr** [comment]
Tests passing! And this tested well for me. @hoangzinh @Beamanator this is ready for another review!

File: tests/unit/ReportUtilsTest.js [hunk 10]
```diff
@@ -410,24 +410,27 @@ describe('ReportUtils', () => {
                 });
             });
 
-            it("it is a non-open expense report tied to user's own paid policy expense chat", () => {
+            it("it is a submitted report tied to user's own policy expense chat and the policy has Instant Submit frequency", () => {
```
Comment (youssef-lr) [inline tests/unit/ReportUtilsTest.js:413]: This is a duplicate test, fixing now

File: src/libs/ReportUtils.ts [hunk 11]
```diff
@@ -5020,6 +5030,17 @@ function canBeAutoReimbursed(report: OnyxEntry<Report>, policy: OnyxEntry<Policy
...
 }
 
+/**
+ * Used from money request actions to decide if we need to build an optimistic money request report.
+   Create a new report if:
+   - we don't have an iouReport set in the chatReport
+   - we have one, but it's waiting on the payee adding a bank account
+   - we have one but we can't add more transactions to it due to: report is approved or settled, or report is processing and policy isn't on Instant submit reporting frequency
+ */
+function shouldBuildOptimisticMoneyRequestReport(existingIOUReport: OnyxEntry<Report> | undefined | null, chatReport: OnyxEntry<Report> | null): boolean {
```
Comment (youssef-lr) [inline src/libs/ReportUtils.ts:5040]: should I rename this to `shouldCreateNewMoneyRequestReport`?

File: src/components/MoneyRequestHeader.tsx (hunk 7 above)
Comment (youssef-lr) [reply src/components/MoneyRequestHeader.tsx:89]: That PR is outdated given the new logic of instant submit. The backend will not throw if the policy has it enabled.

File: src/components/MoneyRequestHeader.tsx (hunk 5 above)
Comment (youssef-lr) [reply src/components/MoneyRequestHeader.tsx:83]: Yup it doesn't. We used `isRequestModifiable` for that before, it doesn't have `isActionOwner`.

**Review by hoangzinh — APPROVED** [review]
Tested again with all workspace types

<details>
<summary>Free WS</summary>


https://github.com/Expensify/App/assets/9639873/ff41a117-337d-4900-9608-490f57266319


https://github.com/Expensify/App/assets/9639873/7886a381-d9fe-4c5b-8985-b7a7b599a655




</details>

<details>
<summary>Collect WS</summary>

https://github.com/Expensify/App/assets/9639873/88a0b32c-063c-4b9d-8d4a-7d480419045b



</details>


<details>
<summary>Collect WS with instant submit</summary>

https://github.com/Expensify/App/assets/9639873/d4f97546-8766-4c14-8bef-7194fa9a780b


https://github.com/Expensify/App/assets/9639873/97922251-7898-4e49-bbd2-b36aa1f0e15b



</details>

**Comment by youssef-lr** [comment]
Awesome thanks! All yours @Beamanator!

**Comment by youssef-lr** [comment]
Not sure why you got assigned @techievivek, Alex is on this :D

File: src/libs/ReportUtils.ts [hunk 12]
```diff
@@ -2957,11 +2972,11 @@ function buildOptimisticExpenseReport(chatReportID: string, policyID: string, pa
     const formattedTotal = CurrencyUtils.convertToDisplayString(storedTotal, currency);
     const policy = getPolicy(policyID);
 
-    const isFree = policy?.type === CONST.POLICY.TYPE.FREE;
+    const isInstantSubmitEnabled = PolicyUtils.isInstantSubmitEnabled(policy);
 
     // Define the state and status of the report based on whether the policy is free or paid
```
[the remark this answers is not in the document]
Comment (youssef-lr) [reply src/libs/ReportUtils.ts:2977]: I think the code is self-explanatory, so we can remove

**Comment by youssef-lr** [comment]
Updated @rlinoz! I think Alex is OOO til Monday he hasn't requested any changes or anything since his last review except that we need to test,  so let's try to get this merged today! 🙏

File: src/libs/ReportUtils.ts [hunk 13]
```diff
@@ -1285,14 +1285,13 @@ function canDeleteReportAction(reportAction: OnyxEntry<ReportAction>, reportID:
         // For now, users cannot delete split actions
         const isSplitAction = reportAction?.originalMessage?.type === CONST.IOU.REPORT_ACTION_TYPE.SPLIT;
 
-        if (isSplitAction || isSettled(String(reportAction?.originalMessage?.IOUReportID)) || (!isEmptyObject(report) && isReportApproved(report))) {
```
Comment (youssef-lr) [inline src/libs/ReportUtils.ts:1273]: I removed this logic as it's already handled in `canAddOrDeleteTransactions`

**Review by rlinoz — CHANGES_REQUESTED** [review]

File: src/libs/ReportUtils.ts [hunk 14]
```diff
@@ -1255,16 +1255,16 @@ function getChildReportNotificationPreference(reportAction: OnyxEntry<ReportActi
  * - report is a draft
  * - report is a processing expense report and its policy has Instant reporting frequency
  */
-function canAddTransactionsToMoneyRequest(report: OnyxEntry<Report>): boolean {
-    if (!isIOUReport(report) && !isExpenseReport(report)) {
+function canAddOrDeleteTransactions(moneyRequestReport: OnyxEntry<Report>): boolean {
+    if (!isMoneyRequest(moneyRequestReport)) {
```
[the remark this answers is not in the document]
Comment (rlinoz) [reply src/libs/ReportUtils.ts:1259]: This might fix the failing tests as well

[the remark this answers is not in the document]
Comment (youssef-lr) [reply src/libs/ReportUtils.ts:1259]: woops just a mistake, will update

[the remark this answers is not in the document]
Comment (youssef-lr) [reply src/libs/ReportUtils.ts]: >if (!isMoneyRequest(moneyRequestReport) && !isMoneyRequestReport(moneyRequestReport)) 

It should only be `isMoneyRequestReport` here, because we can't add transactions to a transaction :D

**Comment by youssef-lr** [comment]
Do you have `bwm` running?

**Comment by rlinoz** [comment]
I do, it is just a little slow sometimes

**Comment by youssef-lr** [comment]
My guess is you tried to submit when the pusher update marking the report as approved wasn't applied yet. Can you try approving, and waiting a bit until you see it was approved, then submit?

**Review by rlinoz — APPROVED** [review]

**Comment by youssef-lr** [comment]
Weird @rlinoz, it gets updated for me in real time:

https://github.com/Expensify/App/assets/9680864/ff638901-4901-4cf3-8797-8cedfb82cae6
